pennylane_lightning/lightning_kokkos/lightning_kokkos.toml

```diff
@@ -112,3 +112,6 @@ dynamic_qubit_management = false
 # in a single execution
 non_commuting_observables = true
 
+# Whether the compilation pass should optimize the first
+# state preparation as a SetState operation.
+skip_initial_state_prep = false
```

Comment:
This should be "true" for both [LK](https://github.com/PennyLaneAI/pennylane-lightning/blob/522f3abd4a3e94eb542e961aae56cb81a93b40e1/pennylane_lightning/lightning_kokkos/lightning_kokkos.py#L472) and [LGPU](https://github.com/PennyLaneAI/pennylane-lightning/blob/522f3abd4a3e94eb542e961aae56cb81a93b40e1/pennylane_lightning/lightning_gpu/lightning_gpu.py#L561)
```suggestion
# Whether the compilation pass should optimize the first
# state preparation as a SetState operation.
skip_initial_state_prep = true
```

Reply:
I'm keeping this one false until the PR in Catalyst is fixed for this. Then I'll update. (This PR will not be merged until the PR in Catalyst is finished)

Reply:
Not sure I understand the reasoning, but as long this is changed before merging.

Reply:
Done.

Reply:
https://github.com/PennyLaneAI/pennylane-lightning/pull/826/commits/efc57527628f89af56eeca7ea71b3278e284d510